web/src/settings_org.ts

```diff
@@ -877,26 +879,14 @@ function set_up_dropdown_widget(
 }
 
 export function set_up_dropdown_widget_for_realm_group_settings(): void {
-    const realm_group_permission_settings = Object.keys(
+    const realm_group_permission_settings = Object.entries(
         realm.server_supported_permission_settings.realm,
     );
 
-    const settings_using_pills_ui = new Set([
-        "can_add_custom_emoji_group",
-        "can_create_groups",
-        "can_create_public_channel_group",
-        "can_create_private_channel_group",
-        "can_delete_any_message_group",
-        "can_delete_own_message_group",
-        "can_manage_all_groups",
-        "can_move_messages_between_channels_group",
-        "can_move_messages_between_topics_group",
-        "create_multiuse_invite_group",
-        "direct_message_initiator_group",
-        "direct_message_permission_group",
-    ]);
-    for (const setting_name of realm_group_permission_settings) {
-        if (settings_using_pills_ui.has(setting_name)) {
+    for (const [setting_name, setting_config] of realm_group_permission_settings) {
+        if (!setting_config.require_system_group) {
+            // For settings that do not require system groups,
+            // we use pills UI.
```

Comment:
I guess there's a bit of an unchecked assertion here that the groups with `require_system_group` here match those that have the matching HTML. Probably not worth worrying about given that this may be temporary.